### ISSUE TITLE
TagHelpers don't work in 2.2 runtime with 3.0 preview sdk

### ISSUE THREAD

**Issue by nattthebear** [aspnet/AspNetCore#6825]
### Describe the bug
Depending on what dotnet sdk I build with, my taghelpers fail to be picked up.

### To Reproduce
1. Using sdk `3.0.100-preview-009812` and with runtime `Microsoft.AspNetCore.App 2.2.1` installed
2. Build and run this project https://github.com/adelikat/tasvideos (I can try to make a smaller reproducing test case if it will help) (Note: project targets `netcoreapp2.2`)
3. The application will run, but the custom TagHelpers will not be picked up

### Expected behavior
Repeating the same steps, but building with sdk `2.2.102`, the TagHelpers work.

### Additional context
```
C:\Develop\TasVideos> dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.2.102
 Commit:    96ff75a873

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.102\

Host (useful for support):
  Version: 3.0.0-preview-27122-01
  Commit:  00c5c8bc40

.NET Core SDKs installed:
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.2.102 [C:\Program Files\dotnet\sdk]
  3.0.100-preview-009812 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview-18579-0056 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview-27122-01 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-alpha-27128-4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by pranavkm** [comment]
@nattthebear thanks for the issue report. The fix for this should most likely be available as part of the 3.0-preview3 release. In the meanwhile, I'd recommend staying on the 2.2 SDK (using a global.json to specify an exact version) until we get a resolution for this.

**Comment by nattthebear** [comment]
Thanks for the prompt response and resolution!
